fetching additional resources reserved in a lease

AnishReddyRavula opened the pull request:
fetching additional resources reserved in a lease, such as hosts, networks, and devices

command to accept a new --detail flag, enabling users to request all resources reserved in a lease

AnishReddyRavula committed:
fetching additional resources reserved in a lease, such as hosts, networks, and devices

command to accept a new --detail flag, enabling users to request all resources reserved in a lease

Mark-Powers commented on blazarclient/v1/leases.py:
```diff
@@ -32,11 +32,15 @@ def create(self, name, start, end, reservations, events, before_end=None):
         condition.
         """
         resp, body = self.request_manager.get('/leases/%s' % lease_id)
+        if detail and body['lease']:
```
How easily could we parallelize this? This is now 4 requests, which might be fairly slow once deployed.

AnishReddyRavula replied:
https://github.com/ChameleonCloud/python-blazarclient/pull/31/commits/ec1cbe0f9c31581fb33087af102539aac8ef32f5 

Used `ThreadPoolExecutor`

AnishReddyRavula committed:
Parallelize calls to get allocations in lease detail get

Parallelize the calls to hosts_in_lease, networks_in_lease, and devices_in_lease using Python's concurrent.futures module. By submitting these calls to a thread pool for concurrent execution and retrieving the results once the tasks are completed, the performance of the get method is potentially improved.

Mark-Powers reviewed (approved):
Looks great!
